spline_background.py

```diff
@@ -0,0 +1,86 @@
+import numpy as np
+from scipy import interpolate
+
+import scipp as sc
+
+
+def bspline_background(variable, dim):
+    """
+    Use scipy bspline method to create a fitting function to data in `variable`.
+    Knots and splines are evaluated internally.
+
+    Parameters
+    ----------
+        variable: scipp DataArray
+            DataArray container to which values the spline should be applied
+        dim: scipp Dim
+            The dimension along which the spline should be calculated
+        output: scipp DataArray
+            DataArray container with the spline
+    """
+    if dim is None:
+        raise ValueError("bspline_background: dimension must be specified.")
+    if not isinstance(dim, sc.Dim):
+        raise ValueError("bspline_background: dimension must be of Dim type.")
+
+    x_values = variable.coords[dim].values
+    values = variable.values
+    errors = variable.variances
+    weights = 1.0/errors
+
+    # spline smoothing from
+    # https://docs.scipy.org/doc/scipy/reference/generated/scipy.interpolate.UnivariateSpline.html
+    s = len(values)*np.std(errors)*np.std(errors)
+    # find out the knots and splines.
+    knots, u = interpolate.splprep([x_values, values], s=s, k=5, w=weights)
+    # perform the B-spline interpolation
+    splined = interpolate.splev(u, knots)
+    # splined is a [2][data_length] array for X and Y values
+
+    # cast splined into DataArray type
+    output_x = sc.Variable(dims=[dim.X], values=splined[0])
+    output_y = sc.Variable(dims=[dim.X], values=splined[1], variances=errors)
```

Comment:
In Mantid this output errors appear to be are based on something `gsl` does (but I did not check in detail), here you are just coping them from the input? Is gsl not modifying them?

Reply:
splprep only returns the sum of squared residuals of the spline approximation (single number), from what I found, rather than the array of residuals.

Reply:
So have you observed differences to what Mantid produces? Maybe it would be better to have an output without variances, unless you think copying the input variances is a good default?

Reply:
gsl_multifit_wlinear() does return variances but scipy doesn't. Those of course will be different than input variances on y.
Yes, I think we should remove them altogether - they aren't correct.

Reply:
:+1: